src/python_bindings/py_util/logging.cpp

```diff
@@ -0,0 +1,50 @@
+#include "logging.h"
+
+#include <unordered_map>
+
+#include <pybind11/pybind11.h>
+#include <spdlog/spdlog.h>
+
+namespace {
+namespace py = pybind11;
+}  // namespace
+
+namespace python_bindings {
+void BindLogging(py::module_& main_module) {
+    auto logging_module = main_module.def_submodule("logging");
+
+    spdlog::set_pattern("[%Y-%m-%d %H:%M:%S.%e] [%^%l%$] %v");
+    spdlog::set_level(spdlog::level::off);  // Default value
+
+    auto set_log_level = [](std::string const& level) {
+        static std::unordered_map<std::string, spdlog::level::level_enum> const kLevels = {
+                {"trace", spdlog::level::trace}, {"debug", spdlog::level::debug},
+                {"info", spdlog::level::info},   {"warn", spdlog::level::warn},
+                {"error", spdlog::level::err},   {"off", spdlog::level::off}};
```

Comment:
Bind the spdlog enum directly: https://pybind11.readthedocs.io/en/stable/classes.html#enumerations-and-internal-types